### PR TITLE
Fix crash on logout while in pending state

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### Description

Steps to reproduce:
1) Turn off internet connection
2) Login
3) Logout
4) The app crashes with the following stacktrace:

```
 java.lang.IllegalStateException: Cannot handle event io.getstream.chat.android.client.clientstate.ClientStateService$ClientStateEvent$DisconnectRequestedEvent@4d413ac while being in inappropriate state io.getstream.chat.android.client.clientstate.ClientState$User$Pending$WithToken@f52f875
```

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [X] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [X] Reviewers added

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientStateService.kt [hunk 1]
```diff
@@ -52,13 +52,7 @@ internal class ClientStateService {
             state<ClientState.User.Pending.WithToken> {
                 onEvent<ClientStateEvent.ConnectedEvent> { state, event -> ClientState.User.Authorized.Connected(event.connectionId, event.user, state.token) }
                 onEvent<ClientStateEvent.DisconnectedEvent> { _, _ -> stay() }
-                onEvent<ClientStateEvent.ConnectedEvent> { state, event ->
-                    ClientState.User.Authorized.Connected(
-                        event.connectionId,
-                        event.user,
-                        state.token
-                    )
-                }
```
Comment (zsmb13) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientStateService.kt:61]: I get the new line being added here, but why is the handling for `ConnectedEvent` being removed?

Comment (bychkovdmitry) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientStateService.kt:61]: Maybe I don't get something about state machines, but the same event is handled above:
```
onEvent<ClientStateEvent.ConnectedEvent> { state, event -> 
      ClientState.User.Authorized.Connected(event.connectionId, event.user, state.token) }
```

Comment (JcMinarro) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientStateService.kt:61]: By some reason it seems taht was duplicated

**Review by JcMinarro — COMMENTED** [review]
Shouldn't we do the same for `state<ClientState.Anonymous.Pending.WithToken>`? :thinking:

**Comment by bychkovdmitry** [comment]
Removed duplication and handled `ClientState.Anonymous.Authorized.Disconnected` event while in `ClientState.Anonymous.Pending.WithToken` state.

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientStateService.kt (hunk 1 above)
Comment (ogkuzmin) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientStateService.kt:61]: Yeah, it was dublicated. Anyway there is a test for such case. It helps us not to break it
(`Given user pending with token state When connected Should move to user connected state`)

**Review by ogkuzmin — APPROVED** [review]
Good job! 👍

**Review by ogkuzmin — COMMENTED** [review]
I think JC is right. It is also good to cover the same case with anonymous user

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**